### PR TITLE
Micro optimizations: cheap checks first

### PR DESCRIPTION
**Pull request by staabm**

### CODE REVIEW

File: packages/VendorLocker/NodeVendorLocker/ClassMethodReturnTypeOverrideGuard.php [hunk 1]
```diff
@@ -48,12 +48,14 @@ public function __construct(
 
     public function shouldSkipClassMethod(ClassMethod $classMethod, Scope $scope): bool
     {
-        // 1. skip magic methods
```
Comment (staabm) [inline packages/VendorLocker/NodeVendorLocker/ClassMethodReturnTypeOverrideGuard.php]: dropped the comments as they did not provide any information, the method call wouldn't already convey on its own

File: packages/VendorLocker/NodeVendorLocker/ClassMethodReturnTypeOverrideGuard.php [hunk 2]
```diff
@@ -80,10 +82,6 @@ public function shouldSkipClassMethod(ClassMethod $classMethod, Scope $scope): b
             return false;
         }
 
-        if ($classMethod->returnType instanceof Node) {
```
Comment (staabm) [inline packages/VendorLocker/NodeVendorLocker/ClassMethodReturnTypeOverrideGuard.php]: moved this check to the top as its the most simple one in this method

File: rules/TypeDeclaration/Rector/ClassMethod/ReturnAnnotationIncorrectNullableRector.php [hunk 3]
```diff
@@ -83,17 +83,17 @@ public function refactorWithScope(Node $node, Scope $scope): ?Node
     {
         $returnType = $node->getReturnType();
 
+        if ($returnType === null) {
+            return null;
+        }
+
         if ($node instanceof ClassMethod && $this->classMethodReturnTypeOverrideGuard->shouldSkipClassMethod(
```
Comment (staabm) [inline rules/TypeDeclaration/Rector/ClassMethod/ReturnAnnotationIncorrectNullableRector.php:90]: no need to invoke the type-overridde-guard, if we already have a return type

**Comment by TomasVotruba** [comment]
Looks good :+1:

**Pull request merged**